src/Features/Core/Portable/Completion/Providers/AbstractEnumCompletionProvider.cs

```diff
@@ -0,0 +1,142 @@
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractEnumCompletionProvider : AbstractSymbolCompletionProvider
+    {
+        private static readonly CompletionItemRules s_rules = CompletionItemRules.Default.WithMatchPriority(MatchPriority.Preselect);
+
+        protected abstract (string displayText, string suffix, string insertionText) GetDefaultDisplayAndSuffixAndInsertionText(ISymbol symbol, SyntaxContext context);
+
+        protected override Task<ImmutableArray<ISymbol>> GetPreselectedSymbolsAsync(SyntaxContext context, int position, OptionSet options, CancellationToken cancellationToken)
+        {
+            var syntaxFacts = context.GetLanguageService<ISyntaxFactsService>();
+            if (syntaxFacts.IsInNonUserCode(context.SyntaxTree, context.Position, cancellationToken))
+                return SpecializedTasks.EmptyImmutableArray<ISymbol>();
+
+            // This providers provides fully qualified names, eg "DayOfWeek.Monday"
+            // Don't run after dot because SymbolCompletionProvider will provide
+            // members in situations like Dim x = DayOfWeek.$$
+            if (context.TargetToken.RawKind == syntaxFacts.SyntaxKinds.DotToken)
+                return SpecializedTasks.EmptyImmutableArray<ISymbol>();
+
+            var typeInferenceService = context.GetLanguageService<ITypeInferenceService>();
+            var enumType = typeInferenceService.InferType(context.SemanticModel, position, objectAsDefault: true, cancellationToken: cancellationToken);
+
+            if (enumType is not { TypeKind: TypeKind.Enum })
+                return SpecializedTasks.EmptyImmutableArray<ISymbol>();
+
+            var hideAdvancedMembers = options.GetOption(RecommendationOptions.HideAdvancedMembers, context.SemanticModel.Language);
+
+            // We'll want to build a list of the actual enum members and all accessible instances of that enum, too
+            var result = enumType.GetMembers().Where(m =>
+                m.Kind == SymbolKind.Field &&
+                ((IFieldSymbol)m).IsConst &&
+                m.IsEditorBrowsable(hideAdvancedMembers, context.SemanticModel.Compilation)).ToImmutableArray();
+
+            return Task.FromResult(result);
+        }
+
+        protected override Task<ImmutableArray<ISymbol>> GetSymbolsAsync(SyntaxContext context, int position,
+            OptionSet options, CancellationToken cancellationToken)
+        {
+            var syntaxFacts = context.GetLanguageService<ISyntaxFactsService>();
+            var node = syntaxFacts.FindTokenOnLeftOfPosition(context.SyntaxTree.GetRoot(cancellationToken), position, includeSkipped: true).Parent;
+            if (syntaxFacts.IsInNonUserCode(context.SyntaxTree, context.Position, cancellationToken) ||
+                syntaxFacts.IsSkippedTokensTrivia(node)) // Note: IsSkippedTokensTrivia implementation is probably buggy
```

Comment:
what is this note about?

Reply:
do we need this check?  do some tests fail if you get rid of it?

Reply:
It is a relict from the port. The `IsSkippedTokensTrivia` can be removed without any influence on the tests.